src/spans/awsSpan.ts

```diff
@@ -232,6 +233,34 @@ export const getAwsServiceData = (requestData, responseData) => {
       return awsParser(requestData, responseData);
   }
 };
+const isJsonContent = (payload, headers) => {
+  return isString(payload) && headers['content-type'] === 'application/json';
+};
+
+export const isContainingSecrets = (body: string) => {
+  const regexes = keyToOmitRegexes();
+  return regexes.some((regex) => regex.test(body));
+};
+
+export const decodeHttpBody = (httpBody: any, hasError: boolean): any | string => {
+  if (isString(httpBody) && httpBody.length < getEventEntitySize(hasError)) {
+    return decode(httpBody);
+  }
+  return httpBody;
+};
+
+const isErrorResponse = (response) => safeGet(response, ['statusCode'], 200) >= 400;
+
+function scrub(payload, headers, sizeLimit: number): string {
+  if (isJsonContent(payload, headers) && isContainingSecrets(payload)) {
+    if (!(payload.length < sizeLimit)) {
```

Comment:
`payload.length < sizeLimit` its more easy then `isContainingSecrets`
```suggestion
    if (isJsonContent(payload, headers) && (!(payload.length < sizeLimit) && isContainingSecrets(payload)) {
```